src/jit/optcse.cpp

```diff
@@ -1712,14 +1710,10 @@ class CSE_Heuristic
 
             // Assert if we used DEBUG_DESTROY_NODE on this CSE exp 
             assert(exp->gtOper != GT_COUNT);
-                
-            /* Ignore the node if it's part of a removed CSE */
-            if  (exp->gtFlags & GTF_DEAD)
-                continue;
+               continue;
```

Comment:
This reads as always do a "continue", which means that the next statement is now unreachable.
That can't be right.